src/hooks/useDisplayDataLostWarning.tsx

```diff
@@ -0,0 +1,47 @@
+import { MessageDialog } from '@/components'
+import React, { useState } from 'react'
+
+type OpenWarningDialogArgs = {
+  cancelCallback?: () => void
+  confirmCallback?: () => void
+}
+
+export const useDisplayDataLostWarning = () => {
```

Comment:
Not so sure about this hook, it's too specific imho. We have issue for handling dialogs (#559), so it will probably need to be removed after merging new dialog handling. Or It can be used as a base for that.

Reply:
so I guess we can stick with that for now

Reply:
Let's keep it as an exploration for now